src/BackgroundProcess/Domain/BackgroundJobQueue.php

```diff
@@ -0,0 +1,29 @@
+<?php
+declare(strict_types=1);
+
+namespace Jetty\BackgroundProcessing\BackgroundProcess\Domain;
+
+/**
+ * Describes the minimum functionality needed for an async request.
+ */
+interface BackgroundJobQueue extends AsyncRequest
+{
+    /**
+     * Push an array of job data to the queue.
+     *
+     * @param array $data Data.
+     */
+    public function pushToQueue(array $data): BackgroundJobQueue;
+
+
+    /**
+     * Save the queue to be processed later.
+     */
+    public function save(): BackgroundJobQueue;
+
+
+    /**
+     * Cancel the background process.
+     */
+    public function cancel(): void;
+}
```

Comment:
These interfaces don't really belong in the domain layer.... If you're writing adapters for interfaces, these really should be application level. Interfaces describe a boundary between lower-level details (such as hardware I/O as in this case) and higher-level application / domain logic.

There really isn't any domain logic here, yet. Although, perhaps, maybe there should be (at some point). I could imagine that some locking domain models could live up here, logic that determines if a process has jammed, etc. And, no, we aren't at that point, yet.

Reply:
Yep. These were some holdovers from my crazy earlier designs. Moved and fixed.